File: Python/Week_1/day2_type_casting.py
# Day 2 - Type Casting in Python

# int to float
x = 10
y = float(x)
print(y, type(y))          # 10.0 <class 'float'>

# float to int (this drops the decimal part, doesn't round!)
a = 9.8
b = int(a)
print(b, type(b))          # 9 <class 'int'>

# string to int
num_str = "25"
num = int(num_str)
print(num + 5, type(num))  # 30 <class 'int'>

# int to string
score = 90
score_str = str(score)
print("Score: " + score_str, type(score_str))

# real-world example: input() always gives string
age = input("Enter your age: ")
age = int(age)              # cast it to int before doing math
print("Next year you'll be:", age + 1)

# common mistake to be aware of
# int("abc")   # this would crash - can't convert non-numeric text to int

File: Java/Week_1/Day2TypeCasting.java
public class Day2TypeCasting {
    public static void main(String[] args) {
        
        // Implicit casting (automatic) - int to double
        int num = 10;
        double numDouble = num;
        System.out.println("Implicit cast: " + numDouble);
        
        // Explicit casting (manual) - double to int
        double price = 99.99;
        int roundedDown = (int) price;
        System.out.println("Explicit cast: " + roundedDown);  // 99, not 100 - it truncates, doesn't round
        
        // String to int
        String numStr = "25";
        int numFromStr = Integer.parseInt(numStr);
        System.out.println("String to int: " + (numFromStr + 5));
        
        // int to String
        int score = 90;
        String scoreStr = String.valueOf(score);
        System.out.println("Score: " + scoreStr);
    }
}